packages/gatsby/src/joi-schemas/joi.ts

```diff
@@ -32,6 +32,7 @@ export const gatsbyConfigSchema: Joi.ObjectSchema<IGatsbyConfig> = Joi.object()
           .replace(/^\/$/, ``)
       )
     ),
+    head: Joi.boolean().default(false),
```

Comment:
Added a new key to the `gatsby-config.js` that is currently disabled by default. To enable (experimental and undocumented) head support, users have to do:

```JS
// gatsby-config.js
module.exports = {
  head: true,
}
```

This will add the react-helmet plugin (if they haven't added it themselves) and render the default meta tags based on the `siteMetadata`.